examples/training/classification/imagenet/basic_training.py

```diff
@@ -197,49 +217,115 @@ def augment(img, label):
         **eval(FLAGS.model_kwargs),
     )
 
-
 """
-Next, we pick an optimizer. Here we use Adam with a constant learning rate.
-Note that learning rate will decrease over time due to the ReduceLROnPlateau callback.
+Optional LR schedule with cosine decay instead of ReduceLROnPlateau
+TODO: Replace with Core Keras LRWarmup when it's released. This is a temporary solution.
 """
 
 
-optimizer = optimizers.SGD(
-    learning_rate=INITIAL_LEARNING_RATE, momentum=0.9, global_clipnorm=10
+def lr_warmup_cosine_decay(
+    global_step, warmup_steps, hold=0, total_steps=0, start_lr=0.0, target_lr=1e-3
+):
+    # Cosine decay
+    # There is no tf.pi :(
+    learning_rate = (
+        0.5
+        * target_lr
+        * (
+            1
+            + tf.cos(
+                tf.constant(np.pi)
+                * (global_step - warmup_steps - hold)
+                / float(total_steps - warmup_steps - hold)
+            )
+        )
+    )
+    warmup_lr = target_lr * (global_step / warmup_steps)
+
```

Comment:
I'm also casting warmup and target lr to float32 here in my run, because otherwise I'm getting errors about incompatible types:

```
warmup_lr = tf.cast(warmup_lr, tf.float32)
target_lr = tf.cast(target_lr, tf.float32)
```